### ISSUE TITLE
about Implementation

### ISSUE THREAD

**Issue by WWWJL** [deepcam-cn/FaceQuality#2]
Hello,I am very interested in your work.and I want to ask some questions, why is the implementation of the code inconsistent with the description in the paper, such as training parameters, learning rate strategy, etc.

**Comment by wjtan99** [comment]
The hyperparameters are changing all the time.  The published parameters may not be same as the ones we used in the paper.

**Comment by WWWJL** [comment]
Thank you for your reply，Recently, I encountered a problem in reconstructing your paper. According to the training hyperparameters in paper, I trained backbone and head from the beginning with MS1M data, and the accuracy of TOP1 reached 98%+. Then, I fixed backbone and trained quality model. However, after trained quality model, it is found that the output quality scores of quality are all 1.



---Original---
From: ***@***.***&gt;
Date: Sun, Jun 20, 2021 11:22 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [deepcam-cn/FaceQuality] about Implementation (#2)




 
The hyperparameters are changing all the time.  The published parameters may not be same as the ones we used in the paper.

**Comment by liurs1990** [comment]
@WWWJL  Is the loss value descending when you train the quality branch?  How do you test the quality score and does it work with our pertained model?

**Comment by WWWJL** [comment]
When training the quality model, the loss keeps decreasing. The test is to use your test script, the quality score is all 1. the quality model is trained with the backbone trained myself(top1 acc 98%+)




---Original---
From: ***@***.***&gt;
Date: Sun, Jun 20, 2021 20:20 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [deepcam-cn/FaceQuality] about Implementation (#2)




 
@WWWJL  Is the loss value descending when you train the quality branch?  How do you test the quality score and does it work with our pertained model?

**Comment by WWWJL** [comment]
The picture is the loss printed when I trained the Quality Model




---Original---
From: ***@***.***&gt;
Date: Sun, Jun 20, 2021 20:20 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [deepcam-cn/FaceQuality] about Implementation (#2)




 
@WWWJL  Is the loss value descending when you train the quality branch?  How do you test the quality score and does it work with our pertained model?

**Comment by liurs1990** [comment]
@WWWJL  Does the quality score work with our pretrained model?

**Comment by WWWJL** [comment]
yes, the quality score work with your pretrained model work!As I said above, the reproduced quality model I trained always has a score of 1 on the test set, and the average value of the quality score of each batch is printed during the training process, which is close to 1.

**Comment by WWWJL** [comment]
> @WWWJL Does the quality score work with our pretrained model?

yes, the quality score work with your pretrained model work!As I said above, the reproduced quality model I trained always has a score of 1 on the test set, and the average value of the quality score of each batch is printed during the training process, which is close to 1.

**Comment by xinlinxie** [comment]
> Thank you for your reply，Recently, I encountered a problem in reconstructing your paper. According to the training hyperparameters in paper, I trained backbone and head from the beginning with MS1M data, and the accuracy of TOP1 reached 98%+. Then, I fixed backbone and trained quality model. However, after trained quality model, it is found that the output quality scores of quality are all 1.
> […](#)
> ---Original--- From: ***@***.***&gt; Date: Sun, Jun 20, 2021 11:22 AM To: ***@***.***&gt;; Cc: ***@***.******@***.***&gt;; Subject: Re: [deepcam-cn/FaceQuality] about Implementation (#2) The hyperparameters are changing all the time. The published parameters may not be same as the ones we used in the paper. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

hello, When I was training the first step, I found that the loss was decreasing, but the accuracy was always 0, can you give me some advice? looking forward your reply~

**Comment by we11d0ne** [comment]
> > 质量分数是否适用于我们的预训练模型？
> 
> 是的，质量分数适用于您的预训练模型工作！如上所述，我训练的重现质量模型在测试集上总是得分为 1，并且在训练过程中打印出每批质量得分的平均值，接近 1。
i have the same mistake , i also have trained  the quality model and the quality score is all 1, have you solved the question?

**Comment by wk100869** [comment]
hello, i have the same mistake , i also have trained the quality model and the quality score is all 1, have you solved the question?

**Comment by we11d0ne** [comment]
I'm sorry, I haven't solved this problem yet. 










-----原始邮件-----
发件人:wk100869 ***@***.***>
发送时间:2024-09-24 10:59:42 (星期二)
收件人: deepcam-cn/FaceQuality ***@***.***>
抄送: we11d0ne ***@***.***>, Comment ***@***.***>
主题: Re: [deepcam-cn/FaceQuality] about Implementation (#2)






hello, i have the same mistake , i also have trained the quality model and the quality score is all 1, have you solved the question?
